internal/provider/resource_gitlab_group_project_file_template.go

```diff
@@ -0,0 +1,116 @@
+		UpdateContext: resourceGitLabGroupProjectFileTemplateCreateOrUpdate,
+		ReadContext:   resourceGitLabGroupProjectFileTemplateRead,
+		DeleteContext: resourceGitLabGroupProjectFileTemplateDelete,
+		// Since this resource updates an in-place resource, importing doesn't make much sense. Simply add the resource
+		// to the config and terraform will overwrite what's already in place and manage it from there.
+		Schema: map[string]*schema.Schema{
+			"group_id": {
+				Description: `The ID of the group that will use the file template project. This group must be the direct
+                parent of the project defined by project_id`,
+				Type: schema.TypeInt,
+
+				// Even though there is no traditional resource to create, leave "ForceNew" as "true" so that if someone
+				// changes a configuration to a different group, the old group gets "deleted" (updated to have a value
+				// of 0).
+				ForceNew: true,
+				Required: true,
+			},
+			"file_template_project_id": {
+				Description: `The ID of the project that will be used for file templates. This project must be the direct
+				child of the project defined by the group_id`,
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+		},
+	}
+})
+
+func resourceGitLabGroupProjectFileTemplateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupId := d.Get("group_id").(int)
+	group, _, err := client.Groups.GetGroup(groupId, nil, gitlab.WithContext(ctx))
+	if err != nil {
+		if is404(err) {
+			log.Printf("[DEBUG] gitlab group %d not found so removing from state", groupId)
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(err)
+	}
+	if group.MarkedForDeletionOn != nil {
+		log.Printf("[DEBUG] gitlab group %s is marked for deletion", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	d.SetId(fmt.Sprintf("%d", group.ID))
+	d.Set("file_template_project_id", group.FileTemplateProjectID)
+
+	return nil
+}
+
+func resourceGitLabGroupProjectFileTemplateCreateOrUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+
+	groupId := d.Get("group_id").(int)
+
+	// Creating the resource means updating the existing group to link the project to the group.
+	options := &gitlab.UpdateGroupOptions{}
+	if d.HasChanges("file_template_project_id") {
+		options.FileTemplateProjectID = gitlab.Int(d.Get("file_template_project_id").(int))
+	}
+
+	_, _, err := client.Groups.UpdateGroup(groupId, options)
+	if err != nil {
+		return diag.Errorf("could not update group after creation %d: %s", groupId, err)
+	}
+	return resourceGitLabGroupProjectFileTemplateRead(ctx, d, meta)
+}
+
+func resourceGitLabGroupProjectFileTemplateDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*gitlab.Client)
+	groupId := d.Get("group_id").(int)
+	options := &gitlab.UpdateGroupOptions{}
+
+	/*
```

Comment:
@PatrickRice-KSC  That's indeed a clever hack 😅 

However, I think we should "workaround" the go-gitlab API by using a request func to alter the body:

```go
_, _, err := client.Groups.UpdateGroup(groupId, nil, gitlab.WithContext(ctx), func(req *retryablehttp.Request) error {
	        hackedOptions := struct {
		        FileTemplateProjectID    *string    `url:"file_template_project_id" json:"file_template_project_id"`
	        }{
		        FileTemplateProjectID:  gitlab.String(""),
	        }
	        body, err := json.Marshal(hackedOptions)
	        if err != nil {
		        return err
	        }
	        if err := req.SetBody(body); err != nil {
		        return err
	        }
		return nil
	})
```

This would intercept the request and set the body with our "hacked" options ... 
Consider it pseudo-code as I haven't tried it ...

Reply:
huh, ok. I'll go ahead and do some testing with this and see if I can get it to work. Thanks!

Reply:
@timofurrer  - Nice code! This worked exactly right the first time 😄 . I refactored the tests using the old method and made sure they passed, then updated this, and tests still pass.